_posts/2016-05-27-microservices.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: He started using microservices and you won't believe what happened....
+author: arto
+excerpt: Easy introduction to microservices
+categories: 
+- Microservices
+tags: 
+- JavaScript
+- Microservices
+- NodeJS
+---
+
+Okay, that was a total clickbait, and a sad one at that. But this article is about microservices, so listen up. Microservices can be a wonderful thing to have, especially in agile projects, but it can also an extremely dangerous tool to use in your architectures. I'm going to try and explain some points here.
+
+To start up, here's simple structure of very simple and typical monolithic application:
+
+![Basic monolithic app architecture](/img/microservices-simple/monolithic.jpg)
+
+When application is simple and has only few responsibilities, this is an okay solution. It's easy enough to maintain with not so many moving parts. Module doesn't get too big and it makes sense to have few database tables in same spot. Troubles occur, however, when this module grows larger. Build times go up, LOC indicators go crazy with hundreds of thousands or millions of lines of code mixed up in same container. If anything goes wrong, all the functionality is offline. Reusability is pretty much zero - of course clever architects may still use libraries here to get a little bit of reusability.
```

Comment:
to have _a_ few database tables in _the_ same spot
mixed up in _the_same container